hs_app_timeseries/receivers.py

```diff
@@ -40,17 +47,42 @@ def pre_add_files_to_resource_handler(sender, **kwargs):
     # if needed more actions can be taken here before content file is added to a TimeSeries resource
     pass
 
+
 @receiver(pre_delete_file_from_resource, sender=TimeSeriesResource)
 def pre_delete_file_from_resource_handler(sender, **kwargs):
-    # if needed more actions can be taken here before content file is deleted from a TimeSeries resource
-    pass
+    # if the deleted file is the sqlite file then reset the 'is_dirty' attribute
+    # for all extracted metadata to False
+    resource = kwargs['resource']
+    del_file = kwargs['file']
+
+    def reset_metadata_elements_is_dirty(elements):
+        for element in elements:
+            element.is_dirty = False
+            element.save()
+
+    # check if it a sqlite file
+    fl_ext, fl_obj_name = get_file_ext_and_obj_name(resource, del_file)
+    if fl_ext == '.sqlite':
+        reset_metadata_elements_is_dirty(resource.metadata.sites.all())
+        reset_metadata_elements_is_dirty(resource.metadata.variables.all())
+        reset_metadata_elements_is_dirty(resource.metadata.methods.all())
+        reset_metadata_elements_is_dirty(resource.metadata.processing_levels.all())
+        reset_metadata_elements_is_dirty(resource.metadata.time_series_results.all())
+        reset_metadata_elements_is_dirty(resource.metadata.cv_variable_types.all())
+        reset_metadata_elements_is_dirty(resource.metadata.cv_variable_names.all())
+        reset_metadata_elements_is_dirty(resource.metadata.cv_speciations.all())
+        reset_metadata_elements_is_dirty(resource.metadata.cv_elevation_datums.all())
+        reset_metadata_elements_is_dirty(resource.metadata.cv_site_types.all())
+        reset_metadata_elements_is_dirty(resource.metadata.cv_method_types.all())
+        reset_metadata_elements_is_dirty(resource.metadata.cv_units_types.all())
+        reset_metadata_elements_is_dirty(resource.metadata.cv_statuses.all())
+        reset_metadata_elements_is_dirty(resource.metadata.cv_aggregation_statistics.all())
```

Comment:
It is necessary to have a dirty flag at the element level so that we can determine which metadata element needs to be written to the SQLite file. Otherwise, even if one element is updated on DB level we have to write all metadata from DB to the SQLite file.